### ISSUE TITLE
The `dropout` operator fails under TO_CPU condition

### ISSUE THREAD

**Issue by tengqm** [FlagOpen/FlagGems#2603]
### Describe the bug

When running under `--ref cpu` mode, the `dropout` operator accuracy test fails.

The detailed output is as follows:

```
shape = (32768,), p = 0.6, dtype = torch.float16

    @pytest.mark.dropout
    @pytest.mark.parametrize("shape", utils.SPECIAL_SHAPES)
    @pytest.mark.parametrize("p", [0.3, 0.6, 0.9])
    @pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
    def test_dropout(shape, p, dtype):
        if flag_gems.vendor_name == "kunlunxin":
            torch.manual_seed(0)
            torch.cuda.manual_seed_all(0)
    
        if cfg.TO_CPU or shape == (1,):
            shape = (32768,)
    
        res_inp = torch.randn(
            shape,
            dtype=dtype,
            device=flag_gems.device,
        )
        ref_inp = utils.to_reference(res_inp)
    
        p = np.float32(p)
        one_minus_p = np.float32(1.0) - p
    
        ref_out = torch.nn.functional.dropout(ref_inp, p, True)
        with flag_gems.use_gems():
            res_out = torch.nn.functional.dropout(res_inp, p, True)
    
        res_out = utils.to_reference(res_out)
        exp_equal = (p * p + one_minus_p * one_minus_p) * res_inp.numel()
        num_equal = torch.sum(torch.isclose(ref_out, res_out)).item()
    
        if cfg.TO_CPU:
            from flag_gems.testing import RESOLUTION
    
            zero_equal = torch.eq(res_out, torch.zeros_like(res_out))
            num_zero = torch.sum(zero_equal).item()
            assert abs(num_zero / res_inp.numel() - p) <= 0.05
            scale_equal = torch.isclose(
                res_out, ref_inp / one_minus_p, rtol=RESOLUTION[dtype]
            )
>           assert torch.all(torch.logical_or(zero_equal, scale_equal))
E           assert tensor(False)
E            +  where tensor(False) = <built-in method all of type object at 0x7fc2e438cf80>(tensor([True, True, True,  ..., True, True, True]))
E            +    where <built-in method all of type object at 0x7fc2e438cf80> = torch.all
E            +    and   tensor([True, True, True,  ..., True, True, True]) = <built-in method logical_or of type object at 0x7fc2e438cf80>(tensor([False,  True, False,  ..., False, False,  True]), tensor([ True, False,  True,  ...,  True,  True, False]))
E            +      where <built-in method logical_or of type object at 0x7fc2e438cf80> = torch.logical_or

tests/test_dropout.py:57: AssertionError
```

**Comment by tengqm** [comment]
This could be caused by random seed unification.
